contrib/agent/src/main/java/io/opencensus/contrib/agent/bootstrap/ContextManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.agent.bootstrap;
+
+/**
+ * {@code ContextManager} provides methods for accessing and manipulating the context from
+ * instrumented bytecode.
+ *
+ * <p>{@code ContextManager} avoids tight coupling with the concrete implementation of the context
+ * by accessing and manipulating the context through the {@link ContextStrategy} interface.
+ *
+ * <p>Both {@link ContextManager} and {@link ContextStrategy} are loaded by the bootstrap
+ * classloader so that they can be used from classes loaded by the bootstrap classloader.
+ * A concrete implementation of {@link ContextStrategy} will be loaded by the system classloader.
+ * This allows for using the same context implementation as the instrumented application.
+ *
+ * <p>{@code ContextManager} is implemented as a static class to allow for easy and fast use from
+ * instrumented bytecode. We cannot use dependency injection for the instrumented bytecode.
+ */
+public final class ContextManager {
+
+  // Not synchronized to avoid any synchronization costs after initialization.
+  // The agent is responsible for initializing this once (through #setContextStrategy) before any
+  // other method of this class is called.
+  private static ContextStrategy contextStrategy;
+
+  private ContextManager() {
+  }
+
+  /**
+   * Sets the concrete strategy for accessing and manipulating the context.
+   *
+   * <p>NB: The agent is responsible for setting the context strategy once before any other method
+   * of this class is called.
+   *
+   * @param contextStrategy the concrete strategy for accessing and manipulating the context
+   */
+  public static void setContextStrategy(ContextStrategy contextStrategy) {
+    if (ContextManager.contextStrategy != null) {
+      throw new IllegalStateException("contextStrategy was already set");
```

Comment:
This seems to be the way how we handle the missing grpc context dependency at runtime. Probably document in the TODO.